lib/chess/move/destination.ex

```diff
@@ -44,6 +44,24 @@ defmodule Chess.Move.Destination do
               Keyword.put(squares, :"#{to}", %Figure{color: color, type: promotion})
             ]
 
+          # valid promotion with opponent
+          x_route == -1 && figure_at_the_end != nil && last_line_for_pion?(color, to) ->
```

Comment:
what if change condition to `(x_route == -1 || x_route == 1) && ...`? or maybe `x_route != 0` ?